### DISCUSSION TITLE
[2024-08-19] Library & Engine Team Update

### DISCUSSION THREAD

**Discussion by GregoryTravis** [enso-org/enso#10840] [Weekly Demo Day]
- James: Initial Tableau Reading Support #10733
- James: Adding support for creating Atoms in expressions #10820
- James: Add Regex link to all regex functions. #10825
- James: Tuning S3_File doc comments. #10832
- Radosław: Various fixes to visualizations #10745
	- Fixed completions on `Snowflake_Connection` and registered it to work with the DB visualization.
	- Added a fallback so types with failing `to_js_object` can still be visualized.
- Radosław: Updating Cloud integration to new directory structure #10762 (waiting for deployment in Cloud)
- Radosław: Adding vectorized implementations to some Column operations #10795
	- Reviewed which operations can be vectorized, implemented vectorized variant for `<=` comparisons where they were not yet vectorized, min and max, and a few more (`is_empty` on mixed storage etc.). Added benchmarks to measure the improvement - the speedup on operations was between 1.75x to up to 60x).
- Radosław: Add retries to tests depending on HTTP to avoid random failures #10815
- Radosław: Fix `Count_Distinct` in Snowflake #10818
- Radosław: Experiment: Reporting test failures as annotations #10821 (PR needs a little bit of cleanup before being ready to review and merge)
- Greg: Mixed Decimal/Float operations throw error or attach warning #10725
- Greg: Column-level lexically-scoped CTE expressions #10826
- Ad: Add newlines option to text cleanse/replace #10761
- Ad: Make sqlserver temp tables work #10781
- Ad: Clean up Text option from read and query #10790
- Ad: Wip/adr/more sql server tests #10810
- Martha: Create node from sort and filter #10704
- Martha: Wip/mk/drive drilldown from enso #10724
- Martha: Add drilldown to XML #10824
- Martha: Sort toolbar rendering for narrow nodes #10831
- Jaroslav: 
    - #10842
    - Package libraries Java code as Java Modules #10714
    - Internal tests run with --disable-private-check #10743
    - Attaching a warning to error is no op #10765
    - Detect outdated launcher #10779
    - https://github.com/oracle/graal/pull/8266
- Hubert: Revert "Import all available libraries in --repl mode (#10746)" #10778 (MacOS OOMs)
- Hubert: Reducing memory leaks in runtime-integration-tests #10793
- Hubert: Cache self argument #10785
- Hubert: Debugging No_Such_Method failures in CI #10750
- Hubert: Dockerize NodeJS-based Ydoc #10811
- Hubert: Optimizing IR copying #10718
- Hubert: Avoid copying IR nodes when no change is performed in passes #10839
- Hubert: Debug Ydoc/LS interaction without making code changes #10687
- Pavel: Store FramePointer in IR #10729 - already gives us [10% speedup](https://github.com/enso-org/enso/pull/10837#issuecomment-2294751914) of _"Hello World"_ startup
- Pavel: Add org.enso.compiler.dumpIr system prop #10740
- Pavel: Flatten object hierarchy of AliasAnalysis metadata so it is accessible from Java #10780
- Pavel: Add --vm.D cmdline option #10798
- Pavel: engine-runner and language-server are separate JPMS modules #10823
- Pavel: Do not run enterprise engine and stdlib benchmarks #10829
- Pavel: Simplify declaration of module path in sbt build #10836
- Dmitry: Disable console logging for project-manager cli commands #10690
- Dmitry: Text editing ranges specified in UTF16 code units #10764
- Dmitry: Ydoc native image #10783

**Comment by radeusgd** [comment]
I wanted to experiment a bit with displaying our test results on the PR in a bit more readable way - using the GH annotations we can get it too look like this:
![image](https://github.com/user-attachments/assets/a791e979-61ed-4b52-bdde-c6a41be4db1d)

I think this should make it a bit easier for us to quickly find simple test failures when working on Libraries.
